force-app/main/default/lwc/eventWithData/__tests__/eventWithData.test.js

```diff
@@ -0,0 +1,96 @@
+const getContactListAdapter = registerLdsTestWireAdapter(getContactList);
+
+describe('c-event-bubbling', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    describe('getContactList @wire data', () => {
+        it('renders two c-contact-list-item elements', () => {
+            // Create initial element
+            const element = createElement('c-event-with-data', {
+                is: EventWithData
+            });
+            document.body.appendChild(element);
+            getContactListAdapter.emit(mockGetContactList);
+            return Promise.resolve().then(() => {
+                const contactListItemEls = element.shadowRoot.querySelectorAll(
+                    'c-contact-list-item'
+                );
+                expect(contactListItemEls.length).toBe(2);
+            });
+        });
+
+        it('renders no c-contact-list-item-bubbling elements when no data', () => {
+            // Create initial element
+            const element = createElement('c-event-event-with-data', {
+                is: EventWithData
+            });
+            document.body.appendChild(element);
+            getContactListAdapter.emit(mockGetContactListNoRecords);
+            return Promise.resolve().then(() => {
+                const contactListItemEls = element.shadowRoot.querySelectorAll(
+                    'c-contact-list-item'
+                );
+                expect(contactListItemEls.length).toBe(0);
+            });
+        });
+    });
+
+    describe('getContactList @wire error', () => {
+        it('shows error panel element', () => {
+            const element = createElement('c-apex-wire-method-to-function', {
+                is: EventWithData
+            });
+            document.body.appendChild(element);
+            getContactListAdapter.error();
+            return Promise.resolve().then(() => {
+                const errorPanelEl = element.shadowRoot.querySelector(
+                    'c-error-panel'
+                );
+                expect(errorPanelEl).not.toBeNull();
+            });
+        });
+    });
+
+    it('shows selected contact data after event', () => {
+        const CONTACT_ID = 99;
+
+        const element = createElement('c-event-with-data', {
+            is: EventWithData
+        });
+        document.body.appendChild(element);
+        getContactListAdapter.emit(mockGetContactList);
+        return Promise.resolve()
+            .then(() => {
+                const contactListItemEls = element.shadowRoot.querySelectorAll(
+                    'c-contact-list-item'
+                );
+                expect(contactListItemEls.length).toBe(2);
+                contactListItemEls[0].dispatchEvent(
```

Comment:
I couldn't figure how to catch this event for validation. Any tips? @trevor-bliss @kevinv11n

Reply:
If you fire a selected event on the `c-contact-list-item` like you are here then it should drop into the `handleSelect` of `eventWithData.js` and update the `selectedContact` prop. After which you can verify the rendered output. 

It looks like that's what you're already trying to do. I don't think you need to catch any events for validation. Is this test not working? Where are you seeing it mess up if so?

Reply:
Yeah, that's not working.

`expect(contactNameEl.textContent).toBe('Amy Taylor');` is failing as no `p` tag gets rendered.

Reply:
Ok I see what's going on. The event payload and the contact data emitted through the `@wire` do not match. The test data has an Id of `"99"` the string and the event has the number 99. Since the component is doing a `===` comparison, these will not match and nothing rendered as the `selectedContact`.

I'm not sure which is correct. Maybe load the app and see if it's a string or boolean and make sure all id values in the tests are that type.

Reply:
🤦‍♂️  - thanks, fixed.